search/engine.go

```diff
@@ -288,17 +287,7 @@ func (e *Engine) GetConfig() *Config {
 	return c
 }
 
-func (e *Engine) updateConfig(databaseName string, configData io.Reader) error {
-	type Table struct {
-		Name  string
-		Table ConfigTable
-	}
-
-	tables := make([]Table, 0)
-	if err := json.NewDecoder(configData).Decode(&tables); err != nil {
-		return err
-	}
-
+func (e *Engine) updateConfig(databaseName string, tables []GenericTable) error {
```

Comment:
What is the difference between updateConfig and updateJSONConfig? Those names don't make it clear.